1.4.0 cannot be compiled on Ubuntu 16.04 with gcc 9.1

Build log
```
16:16:59: Running steps for project SOEM...
16:16:59: Starting: "/home/vincent/cmake-3.12.4-Linux-x86_64/bin/cmake" --build . --target all
[  5%] Building C object CMakeFiles/soem.dir/soem/ethercatbase.c.o
[ 10%] Building C object CMakeFiles/soem.dir/soem/ethercatcoe.c.o
[ 15%] Building C object CMakeFiles/soem.dir/soem/ethercatconfig.c.o
[ 21%] Building C object CMakeFiles/soem.dir/soem/ethercatdc.c.o
[ 26%] Building C object CMakeFiles/soem.dir/soem/ethercateoe.c.o
[ 31%] Building C object CMakeFiles/soem.dir/soem/ethercatfoe.c.o
[ 36%] Building C object CMakeFiles/soem.dir/soem/ethercatmain.c.o
[ 42%] Building C object CMakeFiles/soem.dir/soem/ethercatprint.c.o
[ 47%] Building C object CMakeFiles/soem.dir/soem/ethercatsoe.c.o
[ 52%] Building C object CMakeFiles/soem.dir/osal/linux/osal.c.o
[ 57%] Building C object CMakeFiles/soem.dir/oshw/linux/nicdrv.c.o
[ 63%] Building C object CMakeFiles/soem.dir/oshw/linux/oshw.c.o
In file included from /usr/include/string.h:635,
                 from /mnt/ramdisk/SOEM-1.4.0/oshw/linux/oshw.c:12:
In function ‘strncpy’,
    inlined from ‘oshw_find_adapters’ at /mnt/ramdisk/SOEM-1.4.0/oshw/linux/oshw.c:83:11:
/usr/include/x86_64-linux-gnu/bits/string3.h:126:10: error: ‘__builtin___strncpy_chk’ specified bound depends on the length of the source argument [-Werror=stringop-overflow=]
  126 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/ramdisk/SOEM-1.4.0/oshw/linux/oshw.c: In function ‘oshw_find_adapters’:
/mnt/ramdisk/SOEM-1.4.0/oshw/linux/oshw.c:78:24: note: length computed here
   78 |           string_len = strlen(ids[i].if_name);
      |                        ^~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
CMakeFiles/soem.dir/build.make:205: recipe for target 'CMakeFiles/soem.dir/oshw/linux/oshw.c.o' failed
CMakeFiles/Makefile2:72: recipe for target 'CMakeFiles/soem.dir/all' failed
Makefile:129: recipe for target 'all' failed
make[2]: *** [CMakeFiles/soem.dir/oshw/linux/oshw.c.o] Error 1
make[1]: *** [CMakeFiles/soem.dir/all] Error 2
make: *** [all] Error 2
16:17:01: The process "/home/vincent/cmake-3.12.4-Linux-x86_64/bin/cmake" exited with code 2.
Error while building/deploying project SOEM (kit: 5.13.1 gcc 9)
When executing step "CMake Build"
16:17:01: Elapsed time: 00:03.
```

--- a/oshw/linux/oshw.c
+++ b/oshw/linux/oshw.c
@@ -42,7 +42,6 @@ uint16 oshw_ntohs(uint16 network)
 ec_adaptert * oshw_find_adapters(void)
 {
    int i;
-   int string_len;
    struct if_nameindex *ids;
    ec_adaptert * adapter;
    ec_adaptert * prev_adapter;
@@ -75,15 +74,10 @@ ec_adaptert * oshw_find_adapters(void)
 
       if (ids[i].if_name)
       {
-          string_len = strlen(ids[i].if_name);
-          if (string_len > (EC_MAXLEN_ADAPTERNAME - 1))
-          {
-             string_len = EC_MAXLEN_ADAPTERNAME - 1;
-          }
-          strncpy(adapter->name, ids[i].if_name,string_len);
-          adapter->name[string_len] = '\0';
-          strncpy(adapter->desc, ids[i].if_name,string_len);
-          adapter->desc[string_len] = '\0';
+          strncpy(adapter->name, ids[i].if_name, EC_MAXLEN_ADAPTERNAME);
+          adapter->name[EC_MAXLEN_ADAPTERNAME-1] = '\0';
+          strncpy(adapter->desc, ids[i].if_name, EC_MAXLEN_ADAPTERNAME);
+          adapter->desc[EC_MAXLEN_ADAPTERNAME-1] = '\0';
       }
       else
       {

--- a/oshw/macosx/oshw.c
+++ b/oshw/macosx/oshw.c
@@ -42,7 +42,6 @@ uint16 oshw_ntohs(uint16 network)
 ec_adaptert * oshw_find_adapters(void)
 {
    int i;
-   int string_len;
    struct if_nameindex *ids;
    ec_adaptert * adapter;
    ec_adaptert * prev_adapter;
@@ -70,20 +69,15 @@ ec_adaptert * oshw_find_adapters(void)
          ret_adapter = adapter;
       }
 
-      /* fetch description and name, in Linux we use the same on both */
+      /* fetch description and name, in macosx we use the same on both */
       adapter->next = NULL;
 
       if (ids[i].if_name)
       {
-          string_len = strlen(ids[i].if_name);
-          if (string_len > (EC_MAXLEN_ADAPTERNAME - 1))
-          {
-             string_len = EC_MAXLEN_ADAPTERNAME - 1;
-          }
-          strncpy(adapter->name, ids[i].if_name,string_len);
-          adapter->name[string_len] = '\0';
-          strncpy(adapter->desc, ids[i].if_name,string_len);
-          adapter->desc[string_len] = '\0';
+          strncpy(adapter->name, ids[i].if_name, EC_MAXLEN_ADAPTERNAME);
+          adapter->name[EC_MAXLEN_ADAPTERNAME-1] = '\0';
+          strncpy(adapter->desc, ids[i].if_name, EC_MAXLEN_ADAPTERNAME);
+          adapter->desc[EC_MAXLEN_ADAPTERNAME-1] = '\0';
       }
       else
       {

--- a/oshw/rtems/oshw.c
+++ b/oshw/rtems/oshw.c
@@ -42,7 +42,6 @@ uint16 oshw_ntohs(uint16 network)
 ec_adaptert * oshw_find_adapters(void)
 {
    int i;
-   int string_len;
    struct if_nameindex *ids;
    ec_adaptert * adapter;
    ec_adaptert * prev_adapter;
@@ -70,20 +69,15 @@ ec_adaptert * oshw_find_adapters(void)
          ret_adapter = adapter;
       }
 
-      /* fetch description and name, in Linux we use the same on both */
+      /* fetch description and name, in rtems we use the same on both */
       adapter->next = NULL;
 
       if (ids[i].if_name)
       {
-          string_len = strlen(ids[i].if_name);
-          if (string_len > (EC_MAXLEN_ADAPTERNAME - 1))
-          {
-             string_len = EC_MAXLEN_ADAPTERNAME - 1;
-          }
-          strncpy(adapter->name, ids[i].if_name,string_len);
-          adapter->name[string_len] = '\0';
-          strncpy(adapter->desc, ids[i].if_name,string_len);
-          adapter->desc[string_len] = '\0';
+          strncpy(adapter->name, ids[i].if_name, EC_MAXLEN_ADAPTERNAME);
+          adapter->name[EC_MAXLEN_ADAPTERNAME-1] = '\0';
+          strncpy(adapter->desc, ids[i].if_name, EC_MAXLEN_ADAPTERNAME);
+          adapter->desc[EC_MAXLEN_ADAPTERNAME-1] = '\0';
       }
       else
       {

--- a/oshw/win32/oshw.c
+++ b/oshw/win32/oshw.c
@@ -42,7 +42,6 @@ ec_adaptert * oshw_find_adapters (void)
    ec_adaptert * prev_adapter;
    ec_adaptert * ret_adapter = NULL;
    char errbuf[PCAP_ERRBUF_SIZE];
-   size_t string_len;
 
    /* find all devices */
    if (pcap_findalldevs(&alldevs, errbuf) == -1)
@@ -73,13 +72,8 @@ ec_adaptert * oshw_find_adapters (void)
       adapter->next = NULL;
       if (d->name)
       {
-         string_len = strlen(d->name);
-         if (string_len > (EC_MAXLEN_ADAPTERNAME - 1))
-         {
-            string_len = EC_MAXLEN_ADAPTERNAME - 1;
-         }
-         strncpy(adapter->name, d->name,string_len);
-         adapter->name[string_len] = '\0';
+         strncpy(adapter->name, d->name, EC_MAXLEN_ADAPTERNAME);
+         adapter->name[EC_MAXLEN_ADAPTERNAME-1] = '\0';
       }
       else
       {
@@ -87,13 +81,8 @@ ec_adaptert * oshw_find_adapters (void)
       }
       if (d->description)
       {
-         string_len = strlen(d->description);
-         if (string_len > (EC_MAXLEN_ADAPTERNAME - 1))
-         {
-            string_len = EC_MAXLEN_ADAPTERNAME - 1;
-         }
-         strncpy(adapter->desc, d->description,string_len);
-         adapter->desc[string_len] = '\0';
+         strncpy(adapter->desc, d->description, EC_MAXLEN_ADAPTERNAME);
+         adapter->desc[EC_MAXLEN_ADAPTERNAME-1] = '\0';
       }
       else
       {

